scripts/cookie-auditor/run/index.js

```diff
@@ -0,0 +1,354 @@
+import { pages } from '../configs/main'
+import XLSX from 'xlsx'
+import { chromium, firefox } from 'playwright'
+import isEqual from 'lodash.isequal'
+import path from 'path'
+import fs from 'fs'
+import glob from 'glob'
+
+// =================
+// Main script logic
+// =================
+
+async function run() {
+  console.log('Loading base data into a SheetJS workbook...')
+
+  let baseCookieData = []
+
+  // Check if at least a chromium.xlsx file exists
+  try {
+    await fs.promises.access(path.join(__dirname, '../base-data/chromium.xlsx'))
+    // If it does, create a SheetJS workbook from it
+
+    baseCookieData = glob
+      .sync(path.join(__dirname, '../base-data/*.xlsx'))
+      .map((baseDataFile) => {
+        XLSX.readFileSync(
+          path.join(__dirname, `../base-data/${baseDataFile}.xlsx`)
+        )
+      })
+  } catch {
+    throw Error('No `base-data` file(s) found to compare against! Aborting.')
+  }
+
+  await createNewWorkbooks(pages)
+
+  // Add workbooks generated above to a new array, with browser type as label
+  const workbooks = glob
+    .sync(path.join(__dirname, '../out/cookie-data/*.xlsx'))
+    .map((path) => {
+      const browserTypeRegex = new RegExp(
+        /(?<=cookie-data\/)(.*)\w(?=.xlsx)/,
+        'gi'
+      )
+
+      return {
+        browserType: path.match(browserTypeRegex),
+        newCookieData: XLSX.readFileSync(path),
+      }
+    })
+
+  /**
+   * For each new workbook, compare its data to the base data
+   * and export the differences
+   */
+  for (const { browserType, newCookieData } of workbooks) {
+    await recordCookieDataComparison(baseCookieData, newCookieData, browserType)
+  }
+}
+
+run()
+
+// ==============
+// Functions used
+// ==============
```

Comment:
This thing is a _monster_, I'm aware 😅 Each of these functions can likely be reworked to live in their own files, but the current state of this file reflects my process developing the logic.